calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py

```diff
@@ -147,3 +139,27 @@ def __load_rupture_df(self):
                 self._rupture_df = self._rupture_df.loc[
                     ~self._rupture_df.index.duplicated()
                 ]
+
+    def __load_ims(self):
+        if IMType.pSA in self._im_types:
+            self._ims = set(list(self.branches[0].ims))
```

Comment:
Just a thought to be consistent,
Shall we have the following code instead? so we can safely say `self._ims` is always a list
```python
list(set(list(self.branches[0].ims)))
```

Reply:
But at the same time, the set is still iterable and we will polish this `self._ims` again to be `np.array` so probably not worth casting it.